src/sql/platform/connection/common/connectionConfig.ts

```diff
@@ -277,9 +281,14 @@ export class ConnectionConfig {
 	 * Returns true if connection can be moved to another group
 	 */
 	public canChangeConnectionConfig(profile: ConnectionProfile, newGroupID: string): boolean {
-		let profiles = this.getConnections(true);
-		let existingProfile = find(profiles, p => p.getConnectionInfoId() === profile.getConnectionInfoId()
-			&& p.groupId === newGroupID);
+		let profiles = this.getIConnectionProfileStores(true);
+		let existingProfile = find(profiles, p =>
+			p.providerName === profile.providerName &&
```

Comment:
Seems like the perfect opportunity to add another profile matcher! I'd say generally one that matched the actual connection options only (not the group) would be useful, then you could just add the single additional check here that groupId === newGroupId

Reply:
Yeah unfortunately they're not _actually_ profiles in this state. They're this weird IConnectionProfileStore :/

I could probably look into doing something with it but I've already spent way too much time on this, and honestly this drag & drop needs a major refactor as you pointed out earlier.